### PR TITLE
Windows build support

### PR DESCRIPTION
**Pull request by MaxRis**

### CODE REVIEW

File: recipe/timezone.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--- src\turbodbc_arrow\arrow_result_set.cpp
++++ src\turbodbc_arrow\arrow_result_set.cpp
+@@ -2,6 +2,7 @@
+ 
+ // Somewhere a macro defines BOOL as a constant. This is in conflict with array/type.h
+ #undef BOOL
++#undef timezone
```
Comment (MaxRis) [inline recipe/timezone.patch]: _b_env\include\pyconfig.h at line 230 contains code block with definition of `timezone` for msvc:

```
/* VS 2015 defines these names with a leading underscore */
#if _MSC_VER >= 1900
#define timezone _timezone
#define daylight _daylight
#define tzname _tzname
#endif
```

It causes compilation error for include\arrow\type.h in following code with used `timezone` as variable and method names:

```
explicit TimestampType(TimeUnit::type unit, const std::string& timezone)
      : FixedWidthType(Type::TIMESTAMP), unit_(unit), timezone_(timezone) {}

  Status Accept(TypeVisitor* visitor) const override;
  std::string ToString() const override;
  static std::string name() { return "timestamp"; }

  TimeUnit::type unit() const { return unit_; }
  const std::string& timezone() const { return timezone_; }
```

**Comment by wesm** [comment]
Can you rerender the most recent conda-smithy to include NumPy 1.13 in the build matrix?

@mathmagique @xhochy I presume the Arrow support isn't in the appveyor config for turbodbc yet?

**Comment by MathMagique** [comment]
Yes, it's not in there, mostly because no Windows wheels for arrow 0.4.0 were available at the time of release.

**Comment by wesm** [comment]
OK. We should be able to publish Windows wheels this week for 0.4.1

**Comment by MathMagique** [comment]
That sounds great! In this case, I would prefer having the patch in the turbodbc main repository rather than here. Once the Windows wheel is out, I can activate the automated build and do a turbodbc 2.0.1 release.

**Comment by MaxRis** [comment]
> Can you rerender the most recent conda-smithy to include NumPy 1.13 in the build matrix?
@wesm, rerendering for some reasons adds `CONDA_NPY=113` into `run_docker_build.sh` file, but in this case `ci/circleci` run fails with [this](https://circleci.com/api/v1.1/project/github/conda-forge/turbodbc-feedstock/8/output/13/0?file=true) strange error. So, I've manually removed `CONDA_NPY=113` configuration.
And the most strange, rerendering doesn't generate build matrix for NumPy 1.13 in appveyor.yml. I can see conda search for numpy 1.13.0 in defaults, but `conda-smithy 2.3.1` doesn't add new configuration into build matrix. I can try to extend build matrix manually for this.

**Comment by wesm** [comment]
It's probably because there are not NumPy 1.13 builds for arrow-cpp, pyarrow. These need to be updated and rerendered for Arrow 0.4.1 (just released). Can anyone help with this? Cc @jreback

**Comment by MaxRis** [comment]
@wesm , I can try to update arrow-cpp, pyarrow feedstocks for NumPy 1.13 and Arrow's version "0.4.1"

**Comment by wesm** [comment]
I already started on it, it will take me several hours to carry through. I will report back when I'm done or if i need help

**Comment by MaxRis** [comment]
thank you, @wesm , it seems that rerendering on arrow-cpp-feedstock adds NumPy 1.13 configuration to appveyor build matrix just fine.

**Comment by MaxRis** [comment]
@wesm , thank you! NumPy 1.13 now is added just find by rerendering, all checks passed.

**Comment by MaxRis** [comment]
thank you @xhochy ! Since the patch is trying to make code in include\arrow\type.h be compiled successfully with msvc, can we discuss possibly better solution than just `#undef timezone` in client's cpp code? It hopefully should be possible to put preprocessor macroses directly into include\arrow\type.h of Arrow project. If that sounds fine, I can try to do that.
The `timezone` macro is defined in "_b_env\include\pyconfig.h" only for MSVC. `pyconfig.h` seems to be like some kind of standard file used across python projects, but it's not used in Arrow project looks like, because we didn't have this compilation issue after Arrow was built with msvc.

**Pull request merged**